src/main/java/org/wildfly/security/sasl/scram/ScramSaslClient.java

```diff
@@ -125,16 +126,16 @@ public void init() {
                 b.append(',').append('r').append('=');
                 Random random = secureRandom != null ? secureRandom : ThreadLocalRandom.current();
                 b.append(nonce = ScramUtil.generateNonce(48, random));
-                if (DEBUG) System.out.printf("[C] Client nonce: %s%n", ByteIterator.ofBytes(nonce).hexEncode().drainToString());
+                log.tracef("[C] Client nonce: %s%n", ByteIterator.ofBytes(nonce).hexEncode().drainToString());
                 setNegotiationState(ST_R1_SENT);
-                if (DEBUG) System.out.printf("[C] Client first message: %s%n", ByteIterator.ofBytes(b.toArray()).hexEncode().drainToString());
+                log.tracef("[C] Client first message: %s%n", ByteIterator.ofBytes(b.toArray()).hexEncode().drainToString());
```

Comment:
These still have to be wrapped in some kind of level check (maybe a local trace variable) because these computations are expensive.